ow_regolith/src/RegolithSpawner.cpp

```diff
@@ -310,11 +315,19 @@ void RegolithSpawner::onLinkStatesMsg(const LinkStates::ConstPtr &msg)
   }
 
   auto pose_it = begin(msg->pose) + distance(begin(msg->name), name_it);
-  tf::quaternionMsgToTF(pose_it->orientation, m_scoop_orientation);
+  quaternionMsgToTF(pose_it->orientation, m_scoop_orientation);
 }
 
 void RegolithSpawner::onModDiffVisualMsg(const modified_terrain_diff::ConstPtr& msg)
 {
+  // check if visual terrain modification was caused by the scoop
+  Vector3 scoop_bottom(quatRotate(m_scoop_orientation, m_scoop_down));
+  if (tfDot(scoop_bottom, Vector3(0.0, 0.0, -1.0)) < 0.0)
```

Comment:
#optional: I think you can be more restrictive, 90 degrees or even 80 is unlikely to result in a valid scoop act
```suggestion
  max_potential_scooping = 0.3; // search for a reasonable value
  if (tfDot(scoop_bottom, Vector3(0.0, 0.0, -1.0)) < max_potential_scooping)
```

Reply:
I still prefer the viable dig arc being 180 degrees. There are cases where a dig linear that starts low to the grind will remove terrain at the very start of the first circular trajectory, when the scoop's bottom is 90 degrees relative to the horizontal. I think in this particular case any volume removed by the scoop should still be tracked by the regolith node.

Reply:
`ow_dynamic_terrain` didn't have the gravity vector at hand. anyway like you said this entire thing is workaround. no need to overthink it.